File: Examples/Java/Containers/MongoDB-REST/src/main/java/com/api/mongodb/rest/health/HealthResultEntry.java
package com.api.mongodb.rest.health;

public class HealthResultEntry {
    private String key;
    private boolean healthy;
    private String duration;
    private String status;
    private String error;

    public HealthResultEntry(String key, boolean healthy, long duration, String error) {
        this.key = key;
        this.duration = duration + " ms";
        this.healthy = healthy;
        this.status = healthy ? "healthy" : "unhealthy";
        this.error = error;
    }

    public String getKey() {
        return key;
    }

    public boolean isHealthy() {
        return healthy;
    }

    public String getDuration() {
        return duration;
    }

    public String getStatus() {
        return status;
    }

    public String getError() {
        return error;
    }
}

File: Examples/PHP/Containers/Laravel-MongoDB-REST/app/Models/HealthResultEntry.php
<?php

namespace App\Models;

class HealthResultEntry implements \JsonSerializable
{
    private string $key;
    private string $duration;
    private bool $healthy;
    private string $status;
    private ?string $error;

    public function __construct(string $key, int $duration, string $status, ?string $error = null)
    {
        $this->key = $key;
        $this->duration = $duration . " ms";
        $this->status = strtolower($status);
        $this->healthy = $this->status == 'healthy';
        $this->error = $error;
    }

    public function getKey(): string
    {
        return $this->key;
    }

    public function getDuration(): string
    {
        return $this->duration;
    }

    public function getStatus(): string
    {
        return $this->status;
    }

    public function getHealthy(): bool
    {
        return $this->healthy;
    }

    public function getError(): ?string
    {
        return $this->error;
    }

    public function jsonSerialize(): array
    {
        return [
            'key' => $this->key,
            'duration' => $this->duration,
            'healthy' => $this->healthy,
            'status' => $this->status,
            'error' => $this->error,
        ];
    }
}